drivers/srf02/srf02.c

```diff
@@ -86,22 +78,24 @@ int srf02_init(srf02_t *dev, i2c_t i2c, uint8_t addr)
     return 0;
 }
 
-void srf02_trigger(const srf02_t *dev, srf02_mode_t mode)
+int srf02_trigger(const srf02_t *dev, srf02_mode_t mode)
 {
+    int status;
     /* trigger a new measurement by writing the mode to the CMD register */
     DEBUG("[srf02] trigger new reading\n");
     i2c_acquire(dev->i2c);
-    i2c_write_reg(dev->i2c, dev->addr, REG_CMD, mode);
+    status = i2c_write_reg(dev->i2c, dev->addr, REG_CMD, mode, 0);
     i2c_release(dev->i2c);
+    return status;
 }
 
 uint16_t srf02_read(const srf02_t *dev)
 {
-    uint8_t res[2];
+    uint8_t res[2] = {0xFF, 0xFF};
 
     /* read the results */
     i2c_acquire(dev->i2c);
-    i2c_read_regs(dev->i2c, dev->addr, REG_HIGH, res, 2);
+    i2c_read_regs(dev->i2c, dev->addr, REG_HIGH, res, 2, 0);
```

Comment:
not your fault and I know this is (more) about adapting to the new I2c iface ...

but what if read_regs fails? This function will return garbage then, right? To me it would make more sense to change the function to `int srf02_read(const srf02_t *dev, uint16_t *val)` write result to `val` and return 0 in success and errors accordingly.

However, that would at least be a separate commit, if not a PR -